### DISCUSSION TITLE
Problem using Crow App - '--json' exited with non zero status 1'

### DISCUSSION THREAD

**Discussion by xingz711** [idaholab/moose#16077] [Q&A Getting Started] [answered]
Hi, all, 

I am copying the question from the google group. Problem occurs when I try to use Crow app.

Peacock gives me this:
_Warning
.....projects/Crow/Crow-opt -options_left 0 --json' exited with non zero status 1.
Please make sure your application is built and able to execute the given arguments.

***ERROR***
Invalid option 'SIMPLE' in MooseEnum. Valid options (not case-sensitive) are 'Dirichlet Neumann'_

I test the peacock under Moose_DIR and it works fine with some simple examples. 

Andrew's suggestion:
**cd ~/projects/Crow
./Crow-opt --json**
gives this:
_(moose) [xingz][~/projects3/Crow]> ./Crow-opt --json
*** ERROR ***
Invalid option "SIMPLE" in MooseEnum.  Valid options (not case-sensitive) are "Dirichlet Neumann".

Stack frames: 13
0: libMesh::print_trace(std::ostream&)
1: moose::internal::mooseErrorRaw(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
2: MooseEnum::assign(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
3: MooseEnum::operator=(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
4: MooseEnum::MooseEnum(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool)
5: InputParameters validParams<ElectricBCUO>()
6: decltype (((callValidParamsInner<ElectricBCUO>)(0)),(emptyInputParameters())) moose::internal::callValidParams<ElectricBCUO>()
7: Parser::buildJsonSyntaxTree(JsonSyntaxTree&) const
8: MooseApp::setupOptions()
9: MooseApp::run()
10: ./Crow-opt(+0x51ec) [0x558ecbdc11ec]
11: __libc_start_main
12: ./Crow-opt(+0x54bf) [0x558ecbdc14bf]
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor_

Any suggestions?

**Comment by aeslaughter** [comment, marked as answer]
Somewhere in Crow there is a MooseEnum parameter that is coded to use a value that doesn't exist. I would search for SIMPLE and see what you find.

**Reply by xingz711** [reply]
Hi, Andrew,

Grep 'SIMPLE' and find the term in Crow/src/ElectricBCU

_MooseEnum bc_type ("Dirichlet=0 Neumann", "SIMPLE")_

So I changed it to _MooseEnum bc_type ("Dirichlet=0 Neumann", "Dirichlet Neumann")_.

Now it is giving me this error: _Invalid option "Dirichlet Neumann" in MooseEnum. Valid options (not case-sensitive) are "Dirichlet Neumann"_

I think I need to change it in another way? The content within the brackets looks weird.

**Reply by aeslaughter** [reply]
The part you changed is the default, so pick one of the two, for example.

```
MooseEnum bc_type ("Dirichlet=0 Neumann", "Dirichlet").
```

**Reply by xingz711** [reply]
That fixed the problem! 

Cheers,
Xing

**Reply by SudiptaBiswas** [reply]
@xingz711  I see that this issue has been resolved. I just want to add here that please use the `devel` branch of Crow, it's maintained periodically.
